jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -195,7 +195,7 @@ object XGBoost extends Serializable {
     objective != null && {
       val objStr = objective.toString
       objStr == "classification" || (!objStr.startsWith("reg:") && objStr != "count:poisson" &&
-        objStr != "rank:pairwise")
+        !objStr.startsWith("rank:pairwise"))
```

Comment:
Sorry, I'm missing the difference here. Are there any other built-in rank objectives?

Reply:
rank:ndcg [@n], which is still experimental (doesn't work with hist algo, for one) but gives significantly better results on my datasets

Reply:
Okay, thanks for the explanation. Am I right that you, in fact, want to check `!objStr.startsWith("rank:")`?

Reply:
doh, yes. I'm certain i built and tested with that in my analytics network, but i guess when i redid the patch locally to upload i completely spaced out. New patch incoming.